subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/PmdPlugin.java

```diff
@@ -146,7 +155,15 @@ protected void configureForSourceSet(final SourceSet sourceSet, final Pmd task)
         task.setDescription("Run PMD analysis for " + sourceSet.getName() + " classes");
         task.setSource(sourceSet.getAllJava());
         ConventionMapping taskMapping = task.getConventionMapping();
-        taskMapping.map("classpath", () ->
-            sourceSet.getOutput().plus(sourceSet.getCompileClasspath()));
+        ConfigurationContainer configurations = project.getConfigurations();
+        Configuration compileConfiguration = configurations.getByName(sourceSet.getCompileClasspathConfigurationName());
+        Configuration runtimeConfiguration = configurations.getByName(sourceSet.getRuntimeClasspathConfigurationName());
+        Configuration pmdAuxClasspath = configurations.create(sourceSet.getName() + "PmdAuxClasspath");
+        pmdAuxClasspath.extendsFrom(compileConfiguration, runtimeConfiguration);
```

Comment:
I would rather extend from the configurations these two extend from ... ideally in a lazy fashion of course ...

Reply:
>I would rather extend from the configurations these two extend from ... ideally in a lazy fashion of course ...

Are you talking about `implementation`/`compileOnly`/`api`? It isn't clear to me why it should be better; could you please explain it a bit more? This way a bit of complexity is hidden (e.g. I don't need to know what configurations contribute to the compile classpath or check whether `api` exists).

Reply:
This was me thinking that if we ever get to fully split bucket configurations from resolvable ones, it might make sense to prevent a resolvable from extending another resolvable. But I realize it makes no sense. You are perfectly right that the abstraction is better this way. So my comment was invalid.